api-spec.yaml

```diff
@@ -0,0 +1,179 @@
+consumes:
+- application/json
+info:
+  description: Column - Ansible as a service
+  title: Column
+  version: 0.0.1
+  contact:
+    website: "https://github.com/vmware/column"
+  license:
+    name: "BSD-2"
+    url: "https://github.com/vmware/column/blob/master/LICENSE"
+produces:
+- application/json
+swagger: "2.0"
+schemes:
+- http
+basePath: /v1
+paths:
+  /runs:
+    post:
+      summary: "Create new run that will execute Ansible playbook"
+      description: ""
+      operationId: "createRun"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "Run object that will describe execution context"
+        required: true
+        schema:
+          $ref: "#/definitions/RunPayload"
+      responses:
+        202:
+          description: "Run started successfully"
+          schema:
+            $ref: "#/definitions/Run"
+        403:
+          description: "Access forbidden"
+          schema:
+            $ref: "#/definitions/AccessError"
+        default:
+          description: generic error
+          schema:
+            $ref: "#/definitions/Error"
+    get:
+      summary: "Get all runs in the system"
+      description: ""
+      operationId: "getRuns"
+      responses:
+        200:
+          description: "Returns list of Runs"
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/Run"
+        403:
+          description: "Access forbidden"
+          schema:
+            $ref: "#/definitions/AccessError"
+        default:
+          description: generic error
+          schema:
+            $ref: "#/definitions/Error"
+  /runs/{runId}:
+    get:
+      summary: "Retrieve Run by its ID"
+      description: "Returns single run given its ID"
+      operationID: "getRunByID"
+      parameters:
+      - name: "runId"
+        in: "path"
+        description: "ID of run to return"
+        required: true
+        type: "string"
+        format: "uuid"
+      responses:
+        200:
+          description: "Successful retrieval of Run"
+          schema:
+            $ref: "#/definitions/Run"
+        404:
+          description: "Run does not exist"
+          schema:
+            $ref: "#/definitions/Error"
+        default:
+          description: generic error
+          schema:
+            $ref: "#/definitions/Error"
+  /credentials:
+    get:
+      summary: "Retrieve password by keyfile path"
```

Comment:
It's not a key file.  It's a secret file or credential file.  There is no key in it.